code/modules/antagonists/traitor/medrp/new_objectives.dm

```diff
@@ -0,0 +1,36 @@
+
+
+/obj/effect/proc_holder/spell/self/add_objectives //If all of your objectives are done, you get two more.
+	name = "Request More Objectives"
+	desc = "Grants you more objectives if every objective you have right now is complete."
+	still_recharging_msg = "<span class='notice'>You need to wait for awhile between objective requests.</span>"
+	clothes_req = FALSE
+	antimagic_allowed = TRUE
+	charge_max	= 10 MINUTES
+	charge_counter = 0
+
+/obj/effect/proc_holder/spell/self/add_objectives/cast(list/targets, mob/living/carbon/human/user)
+
+	var/datum/antagonist/traitor/roleplay/antag_datum = user.mind.has_antag_datum(/datum/antagonist/traitor/roleplay)
```

Comment:
it'd be nice if we could turn this into a generic thing that could work with any antag type (with the right subtypes) at some point.